.travis.yml

```diff
@@ -1,50 +1,89 @@
+      addons: {apt: {packages: [cabal-install-1.16, ghc-7.6.3], sources: [hvr-ghc]}}
+    - env: BUILD=cabal CABALVER=1.18 GHCVER=7.8.4
+      addons: {apt: {packages: [cabal-install-1.18, ghc-7.8.4], sources: [hvr-ghc]}}
+    - env: BUILD=cabal CABALVER=1.22 GHCVER=7.10.3
+      addons: {apt: {packages: [cabal-install-1.22, ghc-7.10.3], sources: [hvr-ghc]}}
+    - env: BUILD=cabal CABALVER=1.24 GHCVER=8.0.1
+      addons: {apt: {packages: [cabal-install-1.24, ghc-8.0.1], sources: [hvr-ghc]}}
+    - env: BUILD=stack RESOLVER=lts-2.22 # Last GHC 7.8.x snapshot.
+      addons: &stack-addons
         apt:
-          <<: *apt
-          packages: [ghc-7.6.3, *cabal]
-    - env: GHCVER=7.8.4
+          packages:
+            - libgmp-dev # Stack's GHC depends on this.
+    - env: BUILD=stack RESOLVER=lts-6.4 # Last checked LTS snapshot (GHC 7.10.3).
       addons:
-        apt:
-          <<: *apt
-          packages: [ghc-7.8.4, *cabal]
-    - env: GHCVER=7.10.3
+        <<: *stack-addons
+    - env: BUILD=stack RESOLVER=nightly-2016-06-21 # Last checked nightly snapshot (GHC 8.0.1).
       addons:
-        apt:
-          <<: *apt
-          packages: [ghc-7.10.3, *cabal]
-    - env: GHCVER=8.0.1
+        <<: *stack-addons
+    - env: BUILD=stack RESOLVER=lts # Latest LTS snapshot.
       addons:
-        apt:
-          <<: *apt
-          packages: [ghc-8.0.1, *cabal]
-cache:
-  directories:
-    - '$HOME/.ghc'
-    - '$HOME/.cabal'
+        <<: *stack-addons
+    - env: BUILD=stack RESOLVER=nightly # Latest nightly snapshot.
+      addons:
+        <<: *stack-addons
+  allow_failures:                       # The snapshots `lts` and `nightly` are
+    - env: BUILD=stack RESOLVER=lts     # aliases to the most recent versions.
+    - env: BUILD=stack RESOLVER=nightly # We don't want to fail when a new version
+  fast_finish: true                     # is released, so we list them here.
+
 before_install:
-  - './_test/bootstrap.sh'
+  - export PATH="${TRAVIS_BUILD_DIR}/bin:$PATH"       # For {,fetch-}configlet.
+  - |
+    case "$BUILD" in
+      stack)
+        mkdir -p ${HOME}/bin                          # Create folder for stack.
+        export PATH="${HOME}/bin:$PATH"               # For stack
+        export PATH="${TRAVIS_BUILD_DIR}/_test:$PATH" # For stackalize.
+        ;;
+      cabal)
+        export PATH=/opt/cabal/$CABALVER/bin:/opt/ghc/$GHCVER/bin:$PATH
+        ./_test/bootstrap.sh
+        ;;
+    esac
+
 install:
-  - true
+  - travis_retry fetch-configlet
+  - |
+    set -e
+    if [ "$BUILD" = "stack" ]; then
+      travis_retry curl -L https://www.stackage.org/stack/linux-x86_64 -o pack.tgz
+      tar xzf pack.tgz --wildcards --strip-components=1 -C ${HOME}/bin '*/stack'
+    fi
+
 script:
-  - 'export PATH=/opt/cabal/$CABALVER/bin:/opt/ghc/$GHCVER/bin:$PATH'
-  - 'runhaskell -Wall ./_test/check-exercises.hs'
-  - './bin/fetch-configlet'
-  - './bin/configlet .'
+  - configlet .
+  - |
+    set -e
+    case "$BUILD" in
```

Comment:
In `script:` Travis has a special behavior. If a command fails, Travis will fail everything, but it will not shortcut evaluation, so every entry will be executed.

I used `set -e` and only one command in #162 to avoid that.